src/cli/Cli.ts

```diff
@@ -0,0 +1,180 @@
+        this.optionValues[name] = this.$yargs.option(name, option).argv[name];
+
+        return this;
+    }
+
+    public addOptions(options: Options = {}) {
+        Object.keys(options).forEach(name => {
+            this.addOption(name, options[name]);
+        });
+
+        return this;
+    }
+
+    public addTaskDescription(name: string, desc: string) {
+        this._tasks[name] = {
+            active: false, // Default is, not active
+            ...this._tasks[name],
+            name,
+            desc
+        };
+
+        return this;
+    }
+
+    public addTaskDescriptions(descriptions: TaskDescriptions = {}) {
+        Object.keys(descriptions).forEach(name => {
+            this.addTaskDescription(name, descriptions[name]);
+        });
+
+        return this;
+    }
+
+    public addTask(name: string) {
+        this._tasks[name] = {
+            ...this._tasks[name],
+            name,
+            active: true
+        };
+
+        return this;
+    }
+
+    public showHelp(exitProcess = false) {
+        yargs.getUsageInstance().getCommands().splice(0); // Clean up the current commands
+
+        Object.keys(this._tasks).forEach(name => {
+            const task = this._tasks[name];
+            if (!task.active) return; // The task is not active
+            let taskName = task.name;
+            let taskDesc = task.desc || "";
+
+            if (task.name === "default") {
+                taskName = "\b\b* default\0\0";
+                taskDesc = taskDesc || "The default task";
+            }
+
+            yargs.command(taskName+"\0", "\b"+taskDesc);
+        });
+
+        yargs.showHelp("log");
+        if (exitProcess) process.exit(0);
+
+        return this;
+    }
+
+    public run() {
+        if (!yargs.argv.help) return;
+    
+        this.shutDown(); // Lets shutdown
+        this.showHelp(true); // Show the last help
+
+        return this;
+    }
+
+    public parse(argv = process.argv) {
+        yargs.parse(argv);
+
+        Object.keys(this.optionValues).forEach(name => {
+            this.optionValues[name] = yargs.argv[name];
+        });
+
+        return this;
+    }
+
+    public shutDown() {
+        FuseBoxCLI.initialized = false;
+        Sparky.start = Sparky["$start"] || Sparky.start;
+        Sparky.task = Sparky["$task"] || Sparky.task;
+        delete Sparky["$start"];
+        delete Sparky["$task"];
+        process.removeAllListeners("exit");
+
+        return this;
+    }
+
+    private initYargs() {
+        yargs
+            .reset()
+            .usage("Usage: $0 <task> [options]")
+            .updateStrings({ "Commands:": "Tasks:" })
+            .help(false)
+            .version(false)
+            .option("help", {
+                alias: "h",
+                desc: "Show help",
+                type: "boolean",
+                global: false
+            });
+    }
+
+    private initRunners() {
+        const self = this;
+        process.on("exit", this.run.bind(this));
```

Comment:
Can we use an arrow func here instead?

`process.on("exit", () => this.run());`